### PR TITLE
nixpkgs manual: add rstudio shell.nix example

### PR DESCRIPTION
**Pull request by mulderr**
Provide an example of a self-contained shell.nix to setup an environment with custom R packages and RStudio.

###### Motivation for this change

This could help people new to nix to quickly get started with a working RStudio setup without looking for the right configuration files to modify or people who like to keep all the configuration within each project like myself.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/r-modules/README.md [hunk 1]
```diff
@@ -79,6 +79,30 @@ environment, see `rstudioWrapper`, which functions similarly to
...
+}:
+
+let
+  roverlay = self: super: {
+    rstudioEnv = super.rstudioWrapper.override {
+      packages = with self.rPackages; [ tidyverse ];
+    };
+  };
+
+  pkgs = nixpkgsFunc { overlays = [ roverlay ]; };
```
Comment (peti) [inline pkgs/development/r-modules/README.md]: Why do you define an overlay if all you ever use from the modified package set is just one attribute? That seems unnecessary.

File: pkgs/development/r-modules/README.md [hunk 2]
```diff
@@ -79,6 +79,30 @@ environment, see `rstudioWrapper`, which functions similarly to
 Then like above, `nix-env -f "<nixpkgs>" -iA rstudioEnv` will install
 this into your user profile.
 
+Alternatively, if you would rather not polute your environement or want
```
Comment (peti) [inline pkgs/development/r-modules/README.md]: I think the term "pollute" is inappropriate here. You don't pollute an environment by installing packages into it.

**Review by peti — CHANGES_REQUESTED** [review]

File: pkgs/development/r-modules/README.md (hunk 2 above)
Comment (mulderr) [reply pkgs/development/r-modules/README.md]: Hmm... agreed. Not the best wording. I was trying to refer to the fact with the current example you need to go and modify some configuration files, most likely your config.nix and you may want different things for different projects. I'll update that paragraph to make it clear and use less opinionated terms.

File: pkgs/development/r-modules/README.md (hunk 1 above)
Comment (mulderr) [reply pkgs/development/r-modules/README.md]: Looks like I need to properly read the manual before I submit changes ;) Sorry if it's not up to the standard. I will try to simplify it more. Do you think an example similar this should be in the manual at all?

Comment (mulderr) [reply pkgs/development/r-modules/README.md]: Ok, gotcha. Ineed it was way more complicated than it needed to be. Hope this is better.

**Comment by mulderr** [comment]
Thanks for taking the time to review this!

**Comment by peti** [comment]
You are welcome. And thanks to you for the contribution!

**Pull request merged**